### PR TITLE
Issue #3339566 by vnech: Entity access is ignoring in "Custom content list block" block type

### PR DESCRIPTION
**Pull request by nechai**
## Problem
Anonymous users can see <code>community</code> <code>secret</code> flexible group teasers in "Custom content list block" block:
alt

Should be:
alt

## Solution
Add access check before entities view build.

## Issue tracker
- https://www.drupal.org/project/social/issues/3339566
- https://getopensocial.atlassian.net/browse/PROD-24010

## Theme issue tracker
n/a

## How to test
- [ ] Login as admin
- [ ] Create flexible groups with different visibilities: `public`, `community`, `secret`
- [ ] Create a "**Custom content list block**" on _/block/add/custom_content_list_ to display created groups
- [ ] Place block to any visible region on "**Block layout**" (for example, "**Page title**")
- [ ] As anonymous user visit the page where this block appears

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
_Before changes_:
alt

_After changes_:
alt

## Release notes
Now "Custom content list block" block displays only content (and groups) that user has access to.

## Change Record
n/a

## Translations
n/a

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

File: modules/social_features/social_content_block/src/ContentBuilder.php [hunk 1]
```diff
@@ -162,6 +172,13 @@ static function ($field_name) use ($block_content) {
             ->loadMultiple($entities);
 
           foreach ($entities as $key => $entity) {
+            // @todo Better to move to entity query check but
```
Comment (ribel) [inline modules/social_features/social_content_block/src/ContentBuilder.php:175]: @nechai could you explain more how did you try to make it better and why it didn't work?

**Review by ribel — APPROVED** [review]
Changes looks good to me.
Also tested it on Tugboat and all works as expected.

**LU:**
image


**AN:**
image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

File: modules/social_features/social_content_block/src/ContentBuilder.php (hunk 1 above)
Comment (nkoporec) [reply modules/social_features/social_content_block/src/ContentBuilder.php:175]: Agreed, this needs a bit more explaining on why this change is needed, also release notes are a bit vague. Ideally we would write tests for this, so we don't break it in the  future again.

**Review by nkoporec — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by nechai** [comment]
@nkoporec I agree that would be nice to have behat tests for that feature but as there are a lot of cases that need to be covered (custom content list block can display any content entity in OS) I suggest to create a follow up for tests (currently I don't have a time to work on it).
Does this work for you?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3305-vqyjaxfuxo4jcydyovsuumfrwijdqu1m.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63e0def699029408a80775c9

**Comment by nkoporec** [comment]
@nechai yeah I think creating a follow up is a good plan indeed. Can you address the comments from @ribel  ? Once we clarify that, we can then merge it if everything looks good.

**Comment by nechai** [comment]
@ribel What do you think regarding [this](https://github.com/goalgorilla/open_social/pull/3305#issuecomment-1459649027)?

File: modules/social_features/social_content_block/src/ContentBuilder.php (hunk 1 above)
Comment (nechai) [reply modules/social_features/social_content_block/src/ContentBuilder.php:175]: What I mean here that group visibility doesn't implemented on query level... To achieve this we should rewrite default group [query access handler](https://git.drupalcode.org/project/group/-/blob/8.x-1.x/src/Entity/Group.php#L45). This work require deeper investigation and refinement as can have critical impact on the whole distro.

**Comment by ribel** [comment]
Hi @nechai and @nkoporec 
I've created follow up ticket for adding behat tests and add it to our backlog: https://getopensocial.atlassian.net/browse/PROD-24500
[Explanations](https://github.com/goalgorilla/open_social/pull/3305#discussion_r1129142623) and release notes looks clear to me now.

**Review by nkoporec — APPROVED** [review]

**Comment by nkoporec** [comment]
:cherries:  picked to 11.6.x, 11.7.x, 11.8.x

**Pull request merged**